src/Database/Connection.php

```diff
@@ -331,7 +375,7 @@ public function insert(string $table, array $values, array $types = []): Stateme
     {
         $columns = array_keys($values);
 
-        return $this->newQuery()->insert($columns, $types)
+        return $this->newInsertQuery()->insert($columns, $types)
```

Comment:
`newInsertQuery($columns, $types)` could work too.

Reply:
Yes, appropriate arguments can be added to methods for creating new query instances. Initially I just did what was simplest to replace the existing `newQuery()` calls.

Also I was thinking of moving these methods into a separate `QueryBuilder` class. I don't think the `Connection` class should have the responsibility to creating the query instances.